solutions/fully-configurable/variables.tf

```diff
@@ -0,0 +1,166 @@
+
+variable "parent_enterprise_account_primary_contact_iam_id" {
+  type        = string
+  description = "The IAM id of the parent Enterprise account owner."
+  nullable    = false
+}
+
+variable "enterprise_account_group" {
+  type = list(object({
+    key_name        = string
+    name            = string
+    parent_key_name = optional(string, null)
+    owner_iam_id    = optional(string, null)
+  }))
+  description = "The list of account groups to be created under the parent enterprise account. [Learn more](https://github.com/terraform-ibm-modules/terraform-ibm-enterprise/tree/main/solutions/fully-configurable/DA-account_and_account_group.md)."
+  default     = []
+
+  validation {
+    condition     = length(var.enterprise_account_group) == 0 || length(var.enterprise_account_group) == 1
+    error_message = "The 'enterprise_account_group' list cannot be greater than 1. Received ${length(var.enterprise_account_group)} elements."
+  }
+}
+
+variable "enterprise_account" {
+  type = list(object({
+    key_name               = string
+    name                   = string
+    parent_key_name        = optional(string, null)
+    owner_iam_id           = optional(string, null)
+    add_owner_iam_policies = optional(bool, true)
+    enterprise_iam_managed = optional(bool, true)
+    mfa                    = optional(string, "NONE")
+  }))
+  description = "The list of accounts to be created under the parent enterprise account. [Learn more](https://github.com/terraform-ibm-modules/terraform-ibm-enterprise/tree/main/solutions/fully-configurable/DA-account_and_account_group.md)."
+  default     = []
+
+  validation {
+    condition     = length(var.enterprise_account) == 0 || length(var.enterprise_account) == 1
+    error_message = "The 'enterprise_account' list cannot be greater than 1. Received ${length(var.enterprise_account)} elements."
+  }
+}
+
+##############################################################################
+# trusted profile and access group policy variables
+##############################################################################
+
+variable "trusted_profile_name" {
+  description = "The name of the trusted profile that will be created in the sub account and used to invite users after being assumed by the sub account service ID."
+  type        = string
+  default     = "enable-service-id-to-invite-users"
+}
+
+variable "trusted_profile_description" {
+  description = "The description for the trusted profile that will be created in the sub account and used to invite users after being assumed by the sub account service ID."
+  type        = string
+  default     = "Trusted Profile for sub accounts with required access for inviting users"
+}
+
+variable "access_groups" {
+  description = "Map of access group configurations to create multiple access groups for the newly invited users. [Learn more](https://github.com/terraform-ibm-modules/terraform-ibm-enterprise/tree/main/solutions/fully-configurable/DA-access_groups.md)."
```

Comment:
I would expect this to be a list? Since it supports creating multiple access groups?